src/persist-txn/src/operator.rs

```diff
@@ -219,8 +219,8 @@ where
 
         let mut remap = match empty_to.into_option() {
             Some(empty_to) => DataRemapEntry {
-                physical_upper: empty_to.clone(),
-                logical_upper: empty_to,
+                physical_upper: snap.read_to.unwrap_or(empty_to.clone()),
+                logical_upper: snap.logical_progress,
```

Comment:
If this shard is currently forgotten, then this value needs to be identical to `physical_upper`.